presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -255,7 +256,7 @@ public void retainedBytesForEachPart(BiConsumer<Object, Long> consumer)
     {
         consumer.accept(dictionary, dictionary.getRetainedSizeInBytes());
         consumer.accept(ids, sizeOf(ids));
-        consumer.accept(this, (long) INSTANCE_SIZE);
+        consumer.accept(this, (long) INSTANCE_SIZE + DICTIONARY_ID_SIZE);
```

Comment:
According to the javadoc of this method `consumer visits *each* of the internal data container and accepts the size for it`, although `data container` is a bit vague here it sounds like we need an additional `accept` call with `dictionarySourceId` instead of sticking that right after instance size.